core/src/test/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingServiceSpec.scala

```diff
@@ -166,14 +281,177 @@ class SpendReportingServiceSpec extends AnyFlatSpecLike with TestDriverComponent
+              subAggregatedSpendData.cost shouldBe SpendReportingTestData.SubAggregation.workspace1OtherRowCostRounded.toString
+            }
+          } else {
+            fail(s"unexpected category found in spend results - $subAggregatedSpendData")
+          }
+        }
+
+      } else if (workspaceGoogleProject.equals(SpendReportingTestData.SubAggregation.workspace2.googleProjectId.value)) {
+        withClue(s"cost for ${SpendReportingTestData.SubAggregation.workspace2.toWorkspaceName} was incorrect") {
+          spendData.cost shouldBe SpendReportingTestData.SubAggregation.workspace2TotalCostRounded.toString
+        }
+        subAggregation.spendData.map { subAggregatedSpendData =>
+          if (subAggregatedSpendData.category == Option(TerraSpendCategories.Storage)) {
+            withClue(s"${subAggregatedSpendData.category} category cost for ${SpendReportingTestData.SubAggregation.workspace2.toWorkspaceName} was incorrect") {
+              subAggregatedSpendData.cost shouldBe SpendReportingTestData.SubAggregation.workspace2StorageRowCostRounded.toString
+            }
+          } else if (subAggregatedSpendData.category == Option(TerraSpendCategories.Other)) {
+            withClue(s"${subAggregatedSpendData.category} category cost for ${SpendReportingTestData.SubAggregation.workspace2.toWorkspaceName} was incorrect") {
+              subAggregatedSpendData.cost shouldBe SpendReportingTestData.SubAggregation.workspace2OtherRowCostRounded.toString
+            }
+          } else {
+            fail(s"unexpected category found in spend results - $subAggregatedSpendData")
+          }
+        }
+      } else {
+        fail(s"unexpected workspace found in spend results - $spendData")
+      }
+    }
+  }
+
+  it should "support multiple aggregations" in withDefaultTestDatabase { dataSource: SlickDataSource =>
+    val service = createSpendReportingService(dataSource, tableResult = SpendReportingTestData.SubAggregation.tableResult)
+
+    runAndWait(dataSource.dataAccess.workspaceQuery.createOrUpdate(SpendReportingTestData.Workspace.workspace1))
+    runAndWait(dataSource.dataAccess.workspaceQuery.createOrUpdate(SpendReportingTestData.Workspace.workspace2))
+
+    val reportingResults = Await.result(service.getSpendForBillingProject(testData.billingProject.projectName, DateTime.now().minusDays(1), DateTime.now(), Set(SpendReportingAggregationKeyWithSub(SpendReportingAggregationKeys.Workspace, Option(SpendReportingAggregationKeys.Category)), SpendReportingAggregationKeyWithSub(SpendReportingAggregationKeys.Category))), Duration.Inf)
+    withClue("total cost was incorrect") {
+      reportingResults.spendSummary.cost shouldBe SpendReportingTestData.SubAggregation.totalCostRounded.toString
+    }
+
+    reportingResults.spendDetails.map {
+      case SpendReportingAggregation(SpendReportingAggregationKeys.Workspace, spendDataForWorkspaces) => {
```

Comment:
This block of verification of Workspace~Category is the same is in the previous test point, correct? Could you pull it into a helper method and share it between the tests to remove the duplication? It also makes it easier to read the tests.